### PR TITLE
Feature: Added an option to keep Files running in the background for quicker startup time

### LINKED ISSUE #12614: Feature: Cache windows for quicker startup
### What feature or improvement do you think would benefit Files?

The startup speed is largely limited by WinAppSdk limiations but we can cache windows for a workaround.

### Requirements

- Add experimental setting to enable caching

### Files Version

v2.5.1

### Windows Version

Windows 11 (22H2): 22621

### Comments

_No response_

### PR DESCRIPTION
**Pull request by hishitetsu**
**Resolved / Related Issues**
- [X] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #12614 

**Validation**
How did you test these changes?
- [X] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [X] Are there any other steps that were used to validate these changes?
   1. Turn on "Leave app running in the background when the window is closed" in Advanced settings
   2. Close and restart the app.
   3. The app will launch quicker.

**Known Issues**
 - [X] ~~When only one tab is opened, the tab is not visually selected.~~
image
 - [X] ~~Closing the main window while a property window is still open freezes the property window until the next time the main window is opened.~~
 - [X] ~~The content flashes while it loads.~~

### CODE REVIEW

**Comment by hishitetsu** [comment]
> Closing the main window while a property window is still open freezes the property window until the next time the main window is opened.

One possible solution is to close all property windows together when closing the main window.

**Comment by hishitetsu** [comment]
All known issues were resolved. Now ready for review.

**Comment by Lukiluc29** [comment]
Hello, I just have one last little thing to ask. Is it normal that when I close "Files" and then reopen it, the status center appears? Because before, whether I checked "Open a new tab" or "Continue where you left off," it wouldn't appear when I had performed an action before closing it.

File: src/Files.App/Views/Settings/AdvancedPage.xaml [hunk 1]
```diff
@@ -98,6 +98,17 @@
 				</ToggleSwitch>
 			</local:SettingsBlockControl>
 
+			<!--  Leave App Running  -->
+			<local:SettingsBlockControl Title="{helpers:ResourceString Name=SettingsLeaveAppRunning}" HorizontalAlignment="Stretch">
```
Comment (yaira2) [inline src/Files.App/Views/Settings/AdvancedPage.xaml:102]: Should this be under experimental? If you feel it's stable enough, we can keep it here.

File: src/Files.Core/Services/Settings/IGeneralSettingsService.cs [hunk 2]
```diff
@@ -169,6 +169,11 @@ public interface IGeneralSettingsService : IBaseSettingsService, INotifyProperty
 		/// </summary>
 		bool ShowSendToMenu { get; set; }
 
+		/// <summary>
+		/// Gets or sets a value indicating whether or not to leave app running in the background.
+		/// </summary>
+		bool LeaveAppRunning { get; set; }
```
[the remark this answers is not in the document]
Comment (hishitetsu) [reply src/Files.Core/Services/Settings/IGeneralSettingsService.cs:175]: There is no such service. Should I create a new one?

[the remark this answers is not in the document]
Comment (yaira2) [reply src/Files.Core/Services/Settings/IGeneralSettingsService.cs:175]: Ah right, the other settings on this page aren't tracked by Files. I guess we can leave it as it is for now, we can always create a service in the future if we want to add more settings to this page.

**Comment by hishitetsu** [comment]
> Hello, I just have one last little thing to ask. Is it normal that when I close "Files" and then reopen it, the status center appears? Because before, whether I checked "Open a new tab" or "Continue where you left off," it wouldn't appear when I had performed an action before closing it.

It is true that the behavior is different from when the process is terminated without caching, but I think it doesn't matter.

**Comment by Lukiluc29** [comment]
> It is true that the behavior is different from when the process is terminated without caching, but I think it doesn't matter.

Yes, but it's a bit bothersome because when I close Files, I don't necessarily want to see the status center when I reopen Files, especially when I haven't taken any action upon its reopening.

**Comment by yaira2** [comment]
@Lukiluc29 this is the expected behavior because you aren't actually closing Files, you're leaving it open in the background.

**Comment by Lukiluc29** [comment]
> @Lukiluc29 this is the expected behavior because you aren't actually closing Files, you're leaving it open in the background.

Users tend to close applications after using them, and they might find it disruptive to encounter a status center that they didn't initiate themselves. By removing the status center in this scenario, we can enhance the user's sense of control and provide a smoother and more intuitive interaction with the application.

**Comment by yaira2** [comment]
> Users tend to close applications after using them

Users enabling this setting are opting into this behavior, we might change this in the future but as of now the default setting is to not to leave the app running in the background.

File: src/Files.App/App.xaml.cs [hunk 3]
```diff
@@ -300,6 +301,35 @@ private async void Window_Closed(object sender, WindowEventArgs args)
...
+				MainWindow.Instance.AppWindow.Hide();
+				args.Handled = true;
+
+				// Save and close all tabs
+				SaveSessionTabs();
+				MainPageViewModel.AppInstances.ForEach(tabItem => tabItem.Unload());
+				MainPageViewModel.AppInstances.Clear();
+				await Task.Delay(100);
+
+				// Wait for all properties windows to close
```
Comment (yaira2) [inline src/Files.App/App.xaml.cs:317]: We should be able to do something similar for content dialogs.

**Comment by Lukiluc29** [comment]
> I rather think that notifications should remain even after restarting the app unless the user explicitly clears them. In any case, this is something we can discuss as a separate issue.

Where can we discuss this?

**Comment by hishitetsu** [comment]
> > I rather think that notifications should remain even after restarting the app unless the user explicitly clears them. In any case, this is something we can discuss as a separate issue.
> 
> Where can we discuss this?

You can open a new issue once this PR is merged.

**Comment by mdtauk** [comment]
When the app is in the background, could Files use the system notifications to alert on the completion of on-going tasks?

**Comment by hishitetsu** [comment]
> When the app is in the background, could Files use the system notifications to alert on the completion of on-going tasks?

It stops the thread, so that may be difficult as it is now.
A related issue is already open. #8207

**Comment by yaira2** [comment]
For what its worth, file operations already continue in the background even if the app is fully terminated.

File: src/Files.App/Views/Settings/AdvancedPage.xaml [hunk 4]
```diff
@@ -98,6 +98,17 @@
...
+				<local:SettingsBlockControl.Icon>
+					<FontIcon Glyph="&#xE708;" />
```
Comment (yaira2) [inline src/Files.App/Views/Settings/AdvancedPage.xaml:104]:
```suggestion
					<FontIcon Glyph="&#xE8E6;" />
```

**Review by 0x5bfa — APPROVED** [review]
Looks great to me in terms of the codebase quality!

**Review by yaira2 — APPROVED** [review]
This is a really nice improvement, it's a lot more enjoyable to use Files when you don't have to wait for it to load. LGTM!

**Pull request merged**